search/includes/classes/class-health.php

```diff
@@ -565,4 +570,121 @@ private static function simplified_format_post_diff( $id, $issue ) {
 	private static function get_post_key( $id ) {
 		return sprintf( '%s_%d', 'post', $id );
 	}
+
+	public static function get_index_settings_health_for_all_indexables() {
+		// For each indexable, we want to ensure that the desired index settings match the actual index settings
+		$indexables = \ElasticPress\Indexables::factory()->get_all();
+
+		if ( ! is_array( $indexables ) ) {
+			$message = sprintf( 'Unable to find indexables to check index settings on %s for environment %d', home_url(), FILES_CLIENT_SITE_ID );
+
+			return new \WP_Error( 'no-indexables-found', $message );
+		}
+
+		$unhealthy = array();
+
+		foreach ( $indexables as $indexable ) {
+			$diff = self::get_index_versions_settings_diff_for_indexable( $indexable );
+
+			if ( is_wp_error( $diff ) ) {
+				$unhealthy[ $indexable->slug ] = $diff;
+				
+				continue;
+			}
+			
+			if ( empty( $diff ) ) {
+				continue;
+			}
+
+			$unhealthy[ $indexable->slug ] = $diff;
+		}
+
+		return $unhealthy;
+	}
+
+	public static function get_index_versions_settings_diff_for_indexable( \ElasticPress\Indexable $indexable ) {
+		$search = \Automattic\VIP\Search\Search::instance();
+
+		$versions = $search->versioning->get_versions( $indexable );
+
+		$diff = array();
+
+		foreach ( $versions as $version ) {
+			$version_diff = self::get_index_settings_diff_for_indexable( $indexable, array(
+				'index_version' => $version['number'],
+			) );
+
+			if ( empty( $version_diff ) ) {
+				continue;
+			}
+
+			$diff[] = array(
+				'index_version' => $version['number'],
+				'index_name' => $indexable->get_index_name(),
+				'diff' => $version_diff,
+			);
+		}
+
+		return $diff;
+	}
+
+	public static function get_index_settings_diff_for_indexable( \ElasticPress\Indexable $indexable, $options = array() ) {
+		$search = \Automattic\VIP\Search\Search::instance();
```

Comment:
That's a great idea and would make this function testable (since it would allow us to mock the various functions). This one is a little trickier because it's a static method...will see what I can come up with.

Reply:
This has been done in 797dcd4ca

Reply:
Turns out this breaks unrelated tests (filtering index name for versions)...investigating...